odl/operator/embedding.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2014, 2015 Jonas Adler
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Operators to cast between spaces."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+from odl.space.base_ntuples import FnBase
+from odl.discr.lp_discr import DiscreteLp
+from odl.set.pspace import ProductSpace
+from odl.operator.operator import Operator
+
+__all__ = ('Embedding', 'EmbeddingFnInFn', 'EmbeddingPowerSpaceInFn',
+           'EmbeddingFnInPowerSpace')
+
+
+class Embedding(Operator):
+    """An operators to cast between spaces."""
+
+    # TODO: is this needed?
+
+    def __init__(self, origin, target):
+        super().__init__(origin, target)
+
+
+class EmbeddingFnInFn(Embedding):
+    """ Embed a `FnBase` space into another """
+
+    def __init__(self, origin, target):
+        assert isinstance(origin, FnBase)
+        assert isinstance(target, FnBase)
```

Comment:
I'm wondering if we could replace all our obvious type checks with asserts? At least when the docstring says what you can provide, the error message is easy to relate to the error source.